src/GslCore/PragmaTypes.fs

```diff
@@ -102,144 +102,284 @@ let parsePlatform (args: string list) =
     // a different error.
     | x -> fail (sprintf "platform expected one argument but got %d" x.Length)
 
+/// Represents topology information about contructs
+type Topology =
+    | Linear
+    | Circular
+
+module Topology =
+    [<Literal>]
+    let LinearValue = "linear"
+    
+    [<Literal>]
+    let CircularValue = "circular"
+    
+    [<Literal>]
+    let PragmaName = "topology"
+    
+    let parse : string list -> Result<Topology, string> =
+        function
+        | [arg] ->
+            match arg with
+            | LinearValue -> ok Linear
+            | CircularValue -> ok Circular
+            | _ -> fail (sprintf "Invalid topology '%s'.  Options are 'linear' and 'circular'." arg)
+        | x -> fail (sprintf "topology expected one argument but got %d" x.Length)
+
 /// Pass-through placeholder validator.
 let noValidate _ = ok ()
 
 // Pragma defs that are used internally by the compiler.
 
 let warningPragmaDef =

```

Comment:
There's a ton of diff noise in this file from what looks like a mass-reformatting.  Could you please either revert the reformatting or call out the places where you made substantive changes?

Reply:
sure, in a sec